rts/Sim/Weapons/Weapon.cpp

```diff
@@ -1268,12 +1271,57 @@ float3 CWeapon::GetUnitLeadTargetPos(const CUnit* unit) const
 	return aimPos;
 }
 
+// debug print includes
+//#include<iostream>
+//#include<string>
 
 float3 CWeapon::GetLeadVec(const CUnit* unit) const
 {
-	const float predictTime = GetPredictedImpactTime(unit->pos);
+	float predictTime = GetPredictedImpactTime(unit->pos);
 	const float predictMult = mix(predictSpeedMod, 1.0f, weaponDef->predictBoost);
 
+	if (accurateLeading == true) {
```

Comment:
Consider extracting this chunk to a function that returns the new dist and predictTime